src/System/Commands/DBUpdater/UpdateProjectPopularityCommand.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace App\System\Commands\DBUpdater;
+
+use App\DB\Entity\Project\Program;
+use App\DB\Entity\Project\ProgramLike;
+use App\DB\Entity\Project\Remix\ProgramRemixRelation;
+use App\DB\EntityRepository\Project\ProgramLikeRepository;
+use App\DB\EntityRepository\Project\ProgramRemixRepository;
+use App\DB\EntityRepository\Project\ProgramRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Query\Expr\Join;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class UpdateProjectPopularityCommand extends Command
+{
+  final public const BATCH_SIZE = 1000;
+
+  // Weights for the popularity score computation
+  final public const VIEWS_W = 10;
+  final public const DOWNLOADS_W = 30;
+  final public const REMIXES_W = 45;
+  final public const REACTIONS_W = 15;
+
+  public function __construct(protected EntityManagerInterface $entity_manager, protected ProgramRepository $program_repository, protected ProgramRemixRepository $program_remix_repository, protected ProgramLikeRepository $program_like_repository)
+  {
+    parent::__construct();
+  }
+
+  protected function configure(): void
+  {
+    $this->setName('catrobat:update:popularity')
+      ->setDescription('Updating the popularity score of projects')
+    ;
+  }
+
+  // Compute and update popularity score for every project
+  protected function execute(InputInterface $input, OutputInterface $output): int
+  {
+    $program_count = $this->program_repository->countProjects();
+    $offset = 0;
+    $min_max_values = $this->getMinMaxValues();
+    while ($offset < $program_count) {
+      $programs = $this->program_repository->getProjects(limit: self::BATCH_SIZE, offset: $offset);
+      foreach ($programs as $program) {
+        $popularity = $this->computePopularity($program, $min_max_values);
+        $program->setPopularity($popularity);
+        $this->entity_manager->persist($program);
+      }
+      $offset += self::BATCH_SIZE;
+    }
+    $this->entity_manager->flush();
+    $output->writeln('Popularity scores have been updated');
+
+    return Command::SUCCESS;
```

Comment:
I think this command will run in a timeout, but we can test it out with the next deployment.